pkg/porter/lifecycle.go

```diff
@@ -298,6 +305,31 @@ func (p *Porter) BuildActionArgs(ctx context.Context, installation storage.Insta
 	return args, nil
 }
 
+// ensureVPrefix adds a "v" prefix to the version tag if it's not already there.
+// Version tag should always be prefixed with a "v", see https://github.com/getporter/porter/issues/2886.
+// This is safe because "porter publish" adds a "v", see
+// https://github.com/getporter/porter/blob/17bd7816ef6bde856793f6122e32274aa9d01d1b/pkg/storage/installation.go#L350
+func ensureVPrefix(opts *BundleReferenceOptions) error {
+	if opts._ref == nil {
+		ref, err := cnab.ParseOCIReference(opts.Reference)
```

Comment:
Two things I'm curious about with this:
Is there a case where by doing this we will overwrite or adjust the optsRef set in the closure already?
In what cases would we get to this point and not have the `opts._ref` set?

Reply:
opts.Reference is always set at this point, but opts._ref is only set from `porter explain -r ...` codepath. Also I'm not sure I understand what you mean with by overwrite. The opts.Reference and opts._ref will be overwritten to have a "v" in front if it's already there.

Reply:
decided offline to not update _ref unless it's already non-nil (safest)